c_stubs/xenctrlext_stubs.c

```diff
@@ -195,6 +195,19 @@ CAMLprim value stub_xenctrlext_domain_set_target(value xch,
 	CAMLreturn(Val_unit);
 }
 
+CAMLprim value stub_xenctrlext_physdev_map_pirq(value xch,
+                     value domid,
+                     value irq)
+{
+    CAMLparam3(xch, domid, irq);
+    int pirq = Int_val(irq);
+	caml_enter_blocking_section();
+    int retval = xc_physdev_map_pirq(_H(xch), _D(domid), irq, &pirq);
```

Comment:
I think you need to pass `pirq` as 3rd argument, which is an integer, and not `irq` which is an OCaml value.